### PR TITLE
feat(nix_shell): add symbol to nix-shell module

### LINKED ISSUE #1048: Add Icon to nix environments
## Feature Request
Most other prompt indicators, like Python or Rust, add an "icon" to the prompt (like 🐍 or 🦀 ). Nix, however, has no icon.

#### Describe the solution you'd like
I'd like for Nix envs to have the ❄️ icon, as that is the official logo of the project (c.f. nixos.org).

### PR DESCRIPTION
**Pull request by lovesegfault**
#### Description

Add's a ❄️ icon to the nix-shell module. While at it, changed the default color to blue to be consistent with the nix colorscheme.
#### Motivation and Context

Closes #1048

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Screenshots (if appropriate):
Before:
image

After:
image

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

**Comment by jletey** [comment]
@lovesegfault Might be worth also adding the NerdFont equivalent to the [Presets](https://starship.rs/presets/#nerd-font-symbols) page

Screenshot 2020-04-07 at 09 03 09

File: docs/config/README.md [hunk 1]
```diff
@@ -977,13 +977,14 @@ The module will be shown when inside a nix-shell environment.
...
-| `pure_msg`   | `"pure"`     | Customize the "pure" msg.          |
-| `style`      | `"bold red"` | The style for the module.          |
-| `disabled`   | `false`      | Disables the `nix_shell` module.   |
+| Variable     | Default       | Description                                       |
+| ------------ | ------------- | ------------------------------------------------- |
+| `use_name`   | `false`       | Display the name of the nix-shell.                |
+| `impure_msg` | `"impure"`    | Customize the "impure" msg.                       |
+| `pure_msg`   | `"pure"`      | Customize the "pure" msg.                         |
+| `style`      | `"bold blue"` | The style for the module.                         |
+| `symbol`     | `"❄️  "`       | The symbol used before displaying the shell name. |
```
Comment (andytom) [inline docs/config/README.md:986]: In general we try to keep the `style` and `disabled` variables at the bottom of the table.

```suggestion
| `symbol`     | `"❄️  "`       | The symbol used before displaying the shell name. |
| `style`      | `"bold blue"` | The style for the module.                         |
```

**Comment by andytom** [comment]
> @lovesegfault Might be worth also adding the NerdFont equivalent to the [Presets](https://starship.rs/presets/#nerd-font-symbols) page
> 
> Screenshot 2020-04-07 at 09 03 09

There are a few other modules that are missing from the Presets, I'll create a separate issue for updating the Preset page.

File: docs/config/README.md (hunk 1 above)
Comment (lovesegfault) [reply docs/config/README.md]: Fixed, thanks!

**Pull request merged**